sentinelhub/api/batch_statistical.py

```diff
@@ -0,0 +1,148 @@
+"""
+Module implementing an interface with
+`Sentinel Hub Batch Processing API <https://docs.sentinel-hub.com/api/latest/api/batch-statistical/>`__.
+"""
+import datetime as dt
+import logging
+from dataclasses import dataclass, field
+from typing import Any, Optional, Union
+
+from dataclasses_json import CatchAll, LetterCase, Undefined
+from dataclasses_json import config as dataclass_config
+from dataclasses_json import dataclass_json
+
+from ..type_utils import Json, JsonDict
+from .batch_base import BaseBatchClient, BaseBatchRequest, BatchRequestStatus, BatchUserAction
+from .utils import datetime_config, enum_config, remove_undefined
+
+LOGGER = logging.getLogger(__name__)
+
+BatchStatisticalRequestType = Union[str, dict, "BatchStatisticalRequest"]
+
+
+class SentinelHubBatchStatistical(BaseBatchClient["BatchStatisticalRequest"]):
```

Comment:
Because the service is called Statistical Batch, should we change the name of this class to `SentinelHubStatisticalBatch`?

Reply:
The official docs keep switching between [Statistical Batch](https://docs.sentinel-hub.com/api/latest/reference/#tag/batch_statistical) and [Batch Statistical](https://docs.sentinel-hub.com/api/latest/api/batch-statistical/)

I went with `BatchStatistical` since it occured more often

Reply:
Ah, I see. This is something we could mention to the core team. :see_no_evil: 

IMO, it is nicer if adjective comes before noun in the name but we should follow what the core team decides.

Reply:
Well it think that is the problem, because both of them could be used as the adjective
Batch Statistical (as in statistical done in a batch approach)
Statistical Batch (batch download of statistical nature)

Reply:
After checking with the core team we are keeping the order Batch Statistical.